### DISCUSSION TITLE
All Args Constructor in client generator

### DISCUSSION THREAD

**Discussion by styssi** [quarkiverse/quarkus-openapi-generator#751] [Ideas]
I would like an all All Args Constructor in the client generator. I need this for Unit testing classes that have readOnly attributes. The respone would still be read only, because you don't have setters.

**Comment by hbelmiro** [comment]
I'm not following. The client generates interfaces, which don't have constructors.
Can you elaborate or/and give an example?

**Reply by styssi** [reply]
Attached you can find the openApi (publicdata.json) file and the generated Java file (PublicData.java).

The problem is, that I can't create a PublicData object and give the firstName and lastName content with setter methods. For example I want to set firstName to Tobias for some tests. The reason for no setters is the readOnly tag in the publicdata.json. I do need those setter methods for my Unit Tests. 

As a workaround it would be nice, if the openapi-generator would also generate an AllArgsConstructor inside PublicData.java. That way I could set firstName to Tobias with new PublicData("Tobias", "Miller")

[publicdata.json](https://github.com/user-attachments/files/16391210/publicdata.json)

[PublicData.java.txt](https://github.com/user-attachments/files/16391216/PublicData.java.txt)

**Reply by hbelmiro** [reply]
Oh, you mean in Models.
Yeah, makes sense.
Are you willing to send a PR? I can help with the review.

**Reply by styssi** [reply]
I will give it a try to add the Builder Pattern to model. I will let you know, once I am done.
